pkg/mgmt/volume/volume.go

```diff
@@ -78,6 +78,15 @@ func (c *VolController) syncVol(Vol *apis.LVMVolume) error {
 	var err error
 	// LVM Volume should be deleted. Check if deletion timestamp is set
 	if c.isDeletionCandidate(Vol) {
+		var volExists bool
+		if volExists, err = lvm.CheckVolumeExists(Vol); err != nil {
+			return err
+		}
+		if !volExists {
```

Comment:
If volume was destroyed and remove finalizer failed. In the next call it will return from here. We should still go ahead a try to remove the finalizer.

Reply:
Good catch, forgot to dry run the flow. I'm not sure if we intend to make individual CUD operations in lvm pkg idempotent in future. Moving the check to Destroy volume might simplify the controller logic, but may introduce some inconsistencies with the other funcs (create or resize) in that pkg. Rather than that I've created another method in lvm pkg called `DestroyVolumeIfExists`. Let me know if it looks good.

Reply:
It is better to have this check inside DestroyVolume as it can check if volume is there or not and act accordingly. You brought a good point, create should also be idempotent. If possible can we handle that also in same way we are doing for destroy.

Reply:
Sure, i've added the check in `CreateVolume` as well. One caveat is create volume doesn't ensure the volume of given size. Like if there exists a volume already & its size is not upto what is required as per the create volume request param, we are not resizing or recreating the volume again. If required, probably we can add that logic in future.

Reply:
>  Like if there exists a volume already & its size is not upto what is required as per the create volume request param
That is ok, we expect only one creation request for any volume. And we handle resize differently, the controller does not handle it.

This looks good. I will merge it once travis passes for this PR.